File: Java/1184.java
__________________________________________________________________________________________________
class Solution {
     public int distanceBetweenBusStops(int[] distance, int start, int destination) {

        int index=start;
        int pre=0;
        while (index!=destination){
           pre+=distance[index];
           index=(index+1)%distance.length;
        }
        int last=0;
        while (index!=start){
            last+=distance[index];
            index=(index+1)%distance.length;
        }
        return Math.min(pre,last);
    }
}
__________________________________________________________________________________________________

__________________________________________________________________________________________________

File: Python3/1184.py
__________________________________________________________________________________________________
class Solution:
    def distanceBetweenBusStops(self, distance: List[int], start: int, destination: int) -> int:
        l1 = sum(distance[i] for i in range(start))
        l2 = sum(distance[i] for i in range(destination))
        tot = sum(distance)
        
        cn = max (l1, l2) - min (l1, l2)
        return min (cn, tot - cn)
__________________________________________________________________________________________________

__________________________________________________________________________________________________
